cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -549,7 +550,24 @@ table_with_metadata reader::impl::read_chunk_internal(read_mode mode)
   out_columns.reserve(_output_buffers.size());
 
   // no work to do (this can happen on the first pass if we have no rows to read)
-  if (!has_more_work()) { return finalize_output(out_metadata, out_columns); }
+  if (!has_more_work()) {
+    // Compute number of rows per source if no AST filters
+    if (not _expr_conv.get_converted_expr().has_value()) {
+      // If already empty (empty dataframe case), simply initialize to a list of zeros
+      if (not out_metadata.num_rows_per_source.size()) {
+        out_metadata.num_rows_per_source =
+          std::vector<size_t>(_file_itm_data.num_rows_per_source.size(), 0);
+      }
+      // If this is previously non-empty, simply fill in zeros
```

Comment:
when do we hit this branch?

Reply:
I was thinking if one of the data sources is an empty df (like the following example), we might end up with a zero sized chunk but now realized this actually an error case. Going to remove this branch. Thanks for pointing out.

```python
def test_chunking():
    df = cudf.DataFrame({
            "i64": cudf.Series([1234, None] * 100, dtype="int64")
        })
    empty = cudf.DataFrame({"i64": cudf.Series([], dtype="int64")})

    print(df)
    print(empty)
    buf1 = BytesIO()
    buf2 = BytesIO()

    df.to_parquet(buf1)
    empty.to_parquet(buf2)

    reader = ParquetReader([buf1,buf2,buf1,buf2],chunk_read_limit=100, pass_read_limit=200)
    
    while reader._has_next():
        chunk = reader._read_chunk()
```

Reply:
Funny story, until yesterday it was possible to hit this branch when `skip_rows > row_group_size` in chunked pq reader (in which case we were having zero and negative sized chunks) which ultimately led to segfault or sigabort. This is finally fixed now and we should not hit this branch again!